samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c

```diff
@@ -176,8 +181,93 @@ static void on_location_update(const struct location_event_data * const location
 	}
 }
 
+/**
+ * @brief Receives general device messages from nRF Cloud, checks if they are AT command requests,
+ * and performs them if so, transmitting the modem response back to nRF Cloud.
+ *
+ * Try sending {"appId":"MODEM", "messageType":"CMD", "data":"AT+CGMR"}
+ * in the nRF Cloud Portal Terminal card.
+ *
+ * @param msg - The device message to check.
+ */
+static char at_req_resp_buf[CONFIG_AT_HOST_CMD_MAX_LEN];
+static void handle_at_cmd_requests(const char * const msg) {
+
+	/* Attempt to parse the message as if it is JSON */
+	struct cJSON *msg_obj = cJSON_Parse(msg);
+
+	if (!msg_obj) {
+		/* The message isn't JSON or otherwise couldn't be parsed. */
+		return;
+	}
+
+	/* Check whether the JSON device message is a modem AT command */
+	if (strcmp(cJSON_GetStringValue(cJSON_GetObjectItem(msg_obj, NRF_CLOUD_JSON_APPID_KEY)),
```

Comment:
`cJSON_GetStringValue` could return NULL.

Reply:
Ugh, I forgot that strcmp doesn't do null checking, or at least isn't guaranteed to.

I will reorganize these checks, good catch.

I'd better null check *cmd as well, thinking on it.